src/python/antares_xpansion/config_loader.py

```diff
@@ -508,44 +509,47 @@ def _set_last_simulation_name(self):
         """
         return last simulation name
         """
-        last_dir_path = self.get_last_modified_dir(self.antares_output())
+        self._last_study = self.last_modified_study(self.antares_output())
 
-        if self.step() == "resume":
-            self._simulation_name = last_dir_path
+        if self.step() in ["resume", "sensitivity"] :
+            
+            self._xpansion_simulation_name = self._last_study
+            if self.is_zip(self._last_study):
+                self._xpansion_simulation_name = self._last_study.parent / self._last_study.stem
+                with zipfile.ZipFile(self._last_study, 'r') as output_zip:
+                    output_zip.extractall(self._xpansion_simulation_name)
         else:
-            # Get list of all dirs only in the given directory
-            list_of_zip_filter = Path(self.antares_output()).glob("*.zip")
-
-            # Sort list of files based on last modification time in ascending order
-            list_of_zip = sorted(
-                list_of_zip_filter,
-                key=lambda x: os.path.getmtime(
-                    os.path.join(self.antares_output(), x)),
-            )
-            self._last_zip = list_of_zip[-1]
-            self._simulation_name = self._last_zip.parent / \
-                (self._last_zip.stem+"-Xpansion")
-
-    def is_zip(self, file):
-        filename, ext = os.path.splitext(file)
-        return ext == ".zip"
-
-    def get_last_modified_dir(self, root_dir):
+            self._xpansion_simulation_name = self._last_study.parent / \
+                (self._last_study.stem+"-Xpansion")
+
+    def is_zip(self, study):
+        _, ext = os.path.splitext(study)
+        return ext == ".zip" 
+    
+    def update_last_study_with_sensitivity_results(self):
+        if self.is_zip(self._last_study):
+            os.remove(self._last_study)
+            shutil.make_archive(self._last_study.parent / self._last_study.stem, 'zip', self._xpansion_simulation_name)  
+            if(os.path.exists(self._xpansion_simulation_name)):
+                shutil.rmtree(self._xpansion_simulation_name)
+
+    def is_antares_study(self, study):
```

Comment:
We are looking for output folders / archive here ? If so, `is_antares_study_output` may be a better name